temporian/implementation/numpy/data/event_set.py

```diff
@@ -61,10 +61,29 @@
     DType.FLOAT32: np.float32,
     DType.INT64: np.int64,
     DType.INT32: np.int32,
-    DType.STRING: np.str_,
+    DType.STRING: np.bytes_,
     DType.BOOLEAN: np.bool_,
 }
 
+IndexItemType = Union[int, str, bytes]
+IndexType = Tuple[IndexItemType, ...]
+
+
+def normalize_index_item(x: IndexItemType) -> IndexItemType:
+    if isinstance(x, str):
+        return x.encode()
+    return x
+
+
+def normalize_index(
```

Comment:
would rename to normalize_index_key, since we're now using the word index to refer to an index column/index feature